src/main/java/org/prebid/server/auction/AuctionRequestFactory.java

```diff
@@ -253,11 +254,20 @@ private Site populateSite(Site site, HttpServerRequest request) {
                     final Site.SiteBuilder builder = site == null ? Site.builder() : site.toBuilder();
                     builder.domain(StringUtils.isNotBlank(domain) ? domain : parsedDomain);
                     builder.page(StringUtils.isNotBlank(page) ? page : referer);
+                    if (site == null || site.getExt() == null
```

Comment:
This logic is duplicated in `else` branch, please rework method logic to avoid this duplication